src/passes/OptimizeInstructions.cpp

```diff
@@ -841,6 +757,54 @@ struct OptimizeInstructions
           }
         }
       }
+      if (auto* constTrue = select->ifTrue->dynCast<Const>()) {
+        if (auto* constFalse = select->ifFalse->dynCast<Const>()) {
+          if (select->type == Type::i32 || select->type == Type::i64) {
+            auto trueValue = constTrue->value.getInteger();
+            auto falseValue = constFalse->value.getInteger();
+            if ((trueValue == 1LL && falseValue == 0LL) ||
+                (trueValue == 0LL && falseValue == 1LL)) {
+              Builder builder(*getModule());
+              // canonicalize "expr ? 0 : 1" to "!expr ? 1 : 0"
+              if (trueValue == 0LL) {
+                select->condition = optimizeBoolean(
+                  builder.makeUnary(EqZInt32, select->condition));
+              }
+              Expression* condition = nullptr;
```

Comment:
```suggestion
```

By moving this higher up, and doing the flipping on it, it's clear we are not flipping the entire select.

Reply:
The other `select->condition`s below need to be changed too.